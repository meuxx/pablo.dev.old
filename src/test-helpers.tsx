import { render } from '@testing-library/react'
import { Themed } from './styles/GlobalStyle'
import React from 'react'

const renderComponent: typeof render = (ui, options) => render(ui, { wrapper: Themed, ...options })

const mockComponent = (componentName: string): React.FC => (props) => {
  const { children, ...otherProps } = props
  return (
    <div data-component={componentName} data-props={JSON.stringify(otherProps, null, ' ')}>
      {children}
    </div>
  )
}

export { renderComponent, mockComponent }
