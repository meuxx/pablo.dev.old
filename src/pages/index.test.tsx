import React from 'react'
import { renderComponent, mockComponent } from '../test-helpers'
import IndexPage from './index'

jest.mock('../components/About', () => () => <div />)
jest.mock('../components/Experience', () => () => <div />)
jest.mock('../components/Skills', () => () => <div />)
jest.mock('../components/Layout', () => mockComponent('Layout'))
jest.mock('gatsby-plugin-next-seo', () => ({
  GatsbySeo: mockComponent('GatsbySeo'),
}))

describe('index', () => {
  it('renders without crashing', () => {
    const { container } = renderComponent(<IndexPage />)

    expect(container).toBeDefined()
    expect(container).toMatchSnapshot()
  })
})
