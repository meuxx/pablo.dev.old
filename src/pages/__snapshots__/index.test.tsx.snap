// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders without crashing 1`] = `
.c0 {
  margin-left: 24%;
  max-width: 52em;
  padding: 4em 2em;
  width: calc(100% - 24%);
}

@media screen and (max-width:1280px) {
  .c0 {
    margin-left: 30%;
    max-width: none;
    padding: 6em 3em 3em 3em;
    width: calc(100% - 30%);
  }
}

@media screen and (max-width:980px) {
  .c0 {
    margin: 0;
    padding: 6em 4em;
    width: 100%;
  }
}

@media screen and (max-width:736px) {
  .c0 {
    padding: 2.25em 1.5em 0.25em 1.5em;
  }
}

<div>
  <div
    data-component="Layout"
    data-props="{}"
  >
    <div
      data-component="GatsbySeo"
      data-props="{
 \\"title\\": \\"Pablo Maceda | Back-end developer in London, UK\\",
 \\"description\\": \\"I'm Pablo Maceda, back-end developer offering 5 years of experience, with a solid background as PHP developer and a broad technical skill set\\",
 \\"canonical\\": \\"https://pablo.dev\\",
 \\"openGraph\\": {
  \\"type\\": \\"website\\",
  \\"url\\": \\"https://pablo.dev\\",
  \\"title\\": \\"Pablo Maceda | Back-end developer in London, UK\\",
  \\"site_name\\": \\"Pablo Maceda\\",
  \\"description\\": \\"I'm Pablo Maceda, back-end developer offering 5 years of experience, with a solid background as PHP developer and a broad technical skill set\\",
  \\"images\\": [
   {
    \\"url\\": \\"https://pablo.devtest-file-stub\\"
   }
  ]
 },
 \\"twitter\\": {
  \\"cardType\\": \\"summary_large_image\\"
 },
 \\"metaTags\\": [
  {
   \\"name\\": \\"image\\",
   \\"content\\": \\"https://pablo.devtest-file-stub\\"
  }
 ]
}"
    />
    <main
      class="c0"
    >
      <div />
      <div />
      <div />
    </main>
  </div>
</div>
`;
